internal/api/signup.go

```diff
@@ -184,13 +184,12 @@ func (a *API) Signup(w http.ResponseWriter, r *http.Request) error {
 				}); terr != nil {
 					return terr
 				}
-				flowState, err := models.NewFlowState(params.Provider, params.CodeChallenge, codeChallengeMethod, models.MagicLink)
-				if err != nil {
-					return err
+				flowState, terr := models.NewFlowStateWithUserID(params.Provider, params.CodeChallenge, codeChallengeMethod, models.OTP, &(user.ID))
```

Comment:
Why the pointer thing `&(user.ID)` -- you can probably do it like so:

```suggestion
				flowState, terr := models.NewFlowStateWithUserID(params.Provider, params.CodeChallenge, codeChallengeMethod, models.OTP, &user.ID)
```